core/src/App.svelte

```diff
@@ -375,13 +375,18 @@
     RoutingHelpers.addRouteChangeListener((path, eventDetail) => {
       const { withoutSync, preventContextUpdate } = eventDetail || {};
       const pv = preservedViews;
+
       // TODO: check if bookmarkable modal is interferring here
       if (!isValidBackRoute(pv, path)) {
         preservedViews = [];
         Iframe.removeInactiveIframes(node);
       }
-      for (let i = mfModalList.length; i--; ) {
-        closeModal(i);
+
+      if (!mfModalList.some((item) => item.modalWC && item.mfModal?.displayed)) {
```

Comment:
This condition is a bit hard to read directly. If I can read it correctly this means that:
```
"if in the list of modals there is at least an item that is WC-based and is currently at the top of the stack=displayed"
         don't do anything
else 
        close all modals
```
        

 Maybe the developer had another scenario where they have WC Based wc and displayed and want to close the modals also, not 'do nothing'.
 
I feel like in this way, can always find a failing scenario. For example, now, this fails to close all modals:
```
Luigi.navigation().openAsModal('/test/modalwc', { size:'l', keepPrevious: true});
Luigi.navigation().openAsModal('/test/modalwc2', { size:'m', keepPrevious: true});
Luigi.navigation().openAsModal('/test/overview', { size:'s', keepPrevious: true});
// navigate away
Luigi.navigation().navigate('/test/overview');
```

Question is though, instead of solving scenario after scenario, maybe rethink, what do we want to do when setAnchor is called inside a modal WC. 
Maybe closing all modals and leaving the anchor at the main navigation could also be valid. I am not sure, I'll ask in daily again, since its not a common request.

Reply:
After my latest fix the scenario above will not fail, as all modals are closed after navigating away (separate process). The condition you're mentioning happens after route is updated, so it won't break anything as all modals are closed already